src/test/java/com/github/tomakehurst/wiremock/archunit/UnusedCodeTest.java

```diff
@@ -184,10 +131,63 @@ public void check(JavaMethod javaMethod, ConditionEvents events) {
                               .flatMap(i -> i.getMethods().stream())
                               .anyMatch(hasMatchingNameAndParameters(input))))
               .and()
+              .doNotHaveName("main")
+              .and()
               .haveNameNotContaining("lambda")
+              .and(not(declaredIn(JavaClass.Predicates.ENUMS.or(JavaClass.Predicates.ANNOTATIONS))))
+              .and(
+                  not(
+                      declaredIn(
+                          describe(
+                              "are not declared in Builder",
+                              input -> input.getName().endsWith("Builder")))))
+              .and()
+              .areNotDeclaredIn(com.github.tomakehurst.wiremock.client.WireMock.class)
               .and()
-              .areNotPublic()
+              .areNotDeclaredIn(com.github.tomakehurst.wiremock.core.WireMockConfiguration.class)
+              .and()
+              .areNotDeclaredIn(com.github.tomakehurst.wiremock.junit5.WireMockExtension.class)
+              .and(
+                  not(
+                      describe(
+                          "are not getters",
+                          input ->
+                              input.getParameterTypes().isEmpty()
+                                  && (input.getName().startsWith("get")
+                                      || input.getName().startsWith("is")))))
+              .and(
+                  not(
+                      describe(
+                          "are not builders",
+                          input ->
+                              input.getParameterTypes().size() <= 1
+                                  && input.getOwner().tryGetField(input.getName()).isPresent())))
               .should(beReferencedMethod)
-              .as("should use all non public methods")
+              .as("should use all methods")
```

Comment:
@timtebeek this message change still implies that we're failing on any unused public methods. Am I misunderstanding?

Reply:
That's merely a result of the git diff rendering weirdly in the presence of indentation changes.
The `nonPublicMethodsShouldNotBeUnusedFrozen` isn't really new; it was there before already.
But because the `freeze` was moved from that test to `methodsShouldNotBeUnused` the diff is off.

There's nothing new in terms of what's enforced in this MR; all as it was, with `methodsShouldNotBeUnused` disabled for now. The (now) empty freeze store for `nonPublicMethodsShouldNotBeUnusedFrozen` was repurposed to serve `methodsShouldNotBeUnused`. Hence any confusion; but nothing new is enforced just yet.

Reply:
OK, got it (I think!). I'll merge.

Reply:
Thanks! Glad to see this merged :)